### PR TITLE
Disable auto update at tap command for listing.

### PR DESCRIPTION
**Pull request by rcmdnk**
- [x ] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Currently "Auto Update" is launched at commands of `install`, `upgrade` and `tap`.

Sometime I want to obtain a list of taps, and use `brew tap` command.
But if Auto Update is launched, there are additional lines about Auto Update.

Any installation is done at `brew tap` only when a repository is given,
and Auto Update is necessary only for such a case.

Therefore, this update disables Auto Update  if no additional arguments are given to `brew tap` or listing options of `--list*` are given.

### CODE REVIEW

File: Library/Homebrew/brew.sh [hunk 1]
```diff
@@ -304,7 +305,8 @@ update-preinstall() {
   # Allow auto-update migration now we have a fix in place (below in this function).
   export HOMEBREW_ENABLE_AUTO_UPDATE_MIGRATION="1"
 
-  if [[ "$HOMEBREW_COMMAND" = "install" || "$HOMEBREW_COMMAND" = "upgrade" || "$HOMEBREW_COMMAND" = "tap" ]]
+  echo HOMEBREW_COMMAND_ARGS=${HOMEBREW_COMMAND_ARGS[0]}
```
Comment (scpeters) [inline Library/Homebrew/brew.sh]: I assume this `echo` statement was for debugging; it's probably best to remove it for the PR

**Comment by scpeters** [comment]
I ran into a similar issue where I wanted to untap any non-core taps in a script. I think I was using `brew untap $(brew tap | grep -v homebrew/homebrew-core)`, but the auto-update console messages were causing problems. My solution was to call `brew update` right before that command.

I'm +1 on the idea of this PR.

**Comment by MikeMcQuaid** [comment]
@rcmdnk For future reference it'd be good to push changes to the same PR rather than opening a new one.

**Comment by rcmdnk** [comment]
sorry, I missed that I deleted this PR branch, so that I made a new.

**Pull request closed without merging**